src/Data/String/CodePoints.purs

```diff
@@ -1,7 +1,8 @@
 -- | These functions allow PureScript strings to be treated as if they were
 -- | sequences of Unicode code points instead of their true underlying
 -- | implementation (sequences of UTF-16 code units). For nearly all uses of
--- | strings, these functions should be preferred over the ones in Data.String.
+-- | strings, **these functions should be preferred over the ones in `Data.String`.**
```

Comment:
I'm actually not super keen on the bold here, I think this is OK as-is.

I'd rather not link to the article; a lot of what's in there isn't really relevant to PureScript or this module specifically. I'd prefer to expand the docs here (it hadn't occurred to me before that this would be worth doing, but of course, as you mention, it would be better if the docs here didn't require you to already know what the difference between a code point, a code unit, and a PureScript `Char` was).

Reply:
I guess the relevant part in the article can be summed up in two sentences. Basically UTF16 codepoints are use up two chars.

Reply:
I think it makes sense to go into a bit more detail than that in these docs, but it's probably best to address this separately, as I imagine there will be one or two subtleties which we need to be quite careful with.